lib/mayaUsd/fileio/translators/translatorMaterial.cpp

```diff
@@ -379,16 +379,28 @@ bool UsdMayaTranslatorMaterial::AssignMaterial(
 
         std::string reasonWhyNotPartition;
 
-        const bool validPartition = UsdGeomSubset::ValidateSubsets(
-            faceSubsets, faceCount, UsdGeomTokens->partition, &reasonWhyNotPartition);
+        UsdGeomImageable geom(primSchema.GetPrim());
+        if (UsdGeomSubset::GetFamilyType(geom, UsdShadeTokens->materialBind)
+            != UsdGeomTokens->partition) {
+            TF_WARN(
+                "The family type of family material bind "
+                "on <%s> is not set as a partition.",
+                primSchema.GetPath().GetText());
+        }
+        const bool validPartition = UsdGeomSubset::ValidateFamily(
+            geom, UsdGeomTokens->face, UsdShadeTokens->materialBind, &reasonWhyNotPartition);
```

Comment:
Is the first check above not redundant with the reasonWhyNotPartition? Or are the two checks necessary?

Reply:
It looks like the first check - verifying the FamilyType is partition - is a case that is specific to this function and one which UsdGeomSubset::ValidateFamily doesn't check. I can check with the USD team whether it makes sense to move this to a more general purpose area, but for now this should not be causing redundant warnings.